core/src/main/scala/io/chrisdavenport/shellfish/FilesOs.scala

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright (c) 2024 Typelevel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package io.chrisdavenport.shellfish
+
+import cats.syntax.all.*
+import cats.effect.IO
+
+import fs2.{Stream, Chunk}
+import fs2.interop.scodec.*
+import fs2.io.file.{Files, Path, Flags}
+
+import scodec.{Codec, codecs}
+import scodec.bits.ByteVector
+
+object FilesOs {
+
+  val files = Files[IO]
+
+  // Read operations:
+
+  /**
+   * Reads the contents of the file at the path and returns it as a String.
+   * @param path
+   *   The path to read from
+   * @return
+   *   The file loaded in memory as a String
+   */
+  def read(path: Path): IO[String] = files.readUtf8(path).compile.string
+
+  /**
+   * Reads the contents of the file at the path and returns it as a ByteVector.
+   * @param path
+   *   The path to read from
+   * @return
+   *   The file loaded in memory as a ByteVector
+   */
+  def readBytes(path: Path): IO[ByteVector] =
+    files.readAll(path).compile.to(ByteVector)
+
+  /**
+   * Reads the contents of the file at the path and returns it line by line as a
+   * Vector of Strings.
+   *
+   * @param path
+   *   The path to read from
+   * @return
+   *   The file loaded in memory as a collection of lines of Strings
+   */
+  def readLines(path: Path): IO[Vector[String]] =
```

Comment:
imho, We should prefer `List` over Vector as it's more ubiquitous.
maybe We can add another functions `readAsVector`

Reply:
Agreed. Also I'll specify that is using utf8 and I'll create a second method that accepts a charset, as aforementioned.

Reply:
Counterpoint: `Vector` is better for large collections. Definitely don't use `Seq` or any of the other collection traits as there aren't type class instances for them.